pkg/ccl/changefeedccl/changefeed_test.go

```diff
@@ -3853,6 +3853,56 @@ func TestChangefeedBareAvro(t *testing.T) {
 	cdcTest(t, testFn, feedTestForceSink("kafka"))
 }
 
+func TestChangefeedEnrichedSourceWithNodeAndClusterInfo(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	tc, _, cleanup := startTestCluster(t)
+	defer cleanup()
+	s := tc.Server(0)
+
+	pgURL, cleanup := pgurlutils.PGUrl(t, s.SQLAddr(), t.Name(), url.User(username.RootUser))
+	defer cleanup()
+	pgBase, err := pq.NewConnector(pgURL.String())
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dbWithHandler := gosql.OpenDB(pgBase)
+	defer dbWithHandler.Close()
+
+	ctx := context.Background()
+	sqlDB := sqlutils.MakeSQLRunner(dbWithHandler)
+	sqlDB.Exec(t, `CREATE TABLE foo (i INT PRIMARY KEY)`)
+	sqlDB.Exec(t, `INSERT INTO foo VALUES (0)`)
+
+	clusterID := s.ExecutorConfig().(sql.ExecutorConfig).NodeInfo.LogicalClusterID().String()
+	// clusterName is "" in tests. This is a valid value.
+	clusterName := s.ExecutorConfig().(sql.ExecutorConfig).RPCContext.ClusterName()
+	dbVersion := s.ClusterSettings().Version.ActiveVersion(ctx).String()
+
+	t.Run("kafka", func(t *testing.T) {
+		f := makeKafkaFeedFactory(t, s, dbWithHandler)
+		testFeed := feed(t, f, `CREATE CHANGEFEED FOR foo WITH envelope=enriched, enriched_properties='source', format=json`)
+		defer closeFeed(t, testFeed)
+
+		var jobID int64
+		var nodeName string
+		// Fetching the jobId this way is not supported by the sinkless sink.
+		// In that case we will assert the job_id is 0, so we don't need this query.
+		if _, ok := testFeed.(*sinklessFeed); !ok {
+			sqlDB.QueryRow(t, `SELECT job_id FROM [SHOW JOBS] where job_type='CHANGEFEED'`).Scan(&jobID)
```

Comment:
can't you just do `if ef, ok := testFeed.(cdctest.EnterpriseTestFeed); ok {jobID = ef.JobID()}`?

Reply:
also currently you're forcing a kafka feed which is not sinkless